tests/unit/steps/test_steps.py

```diff
@@ -32,21 +32,36 @@
     compare_step_coroutines,
     is_unit_upgrade_step,
 )
+from cou.utils import app_utils
 
 
-async def mock_coro(*args, **kwargs): ...
+async def mock_coro(a=1, b=2, c=3, *args, **kwargs): ...
+
+
+async def _coro(name: str, force: bool = False, units: list | None = None): ...
 
 
 @pytest.mark.parametrize(
     "coro1, coro2, exp_result",
     [
-        (None, mock_coro(), False),
-        (mock_coro(), None, False),
-        (mock_coro(), mock_coro(1, 2, 3), False),
-        (mock_coro(), mock_coro(arg1=True), False),
-        (mock_coro(), mock_coro(), True),
-        (mock_coro(1, 2, 3, kwarg1=True), mock_coro(1, 2, 3, kwarg1=True), True),
-        (mock_coro(1, 2, kwarg1=3, kwarg2=True), mock_coro(1, 2, 3, True), True),
+        (None, _coro("test/0"), False),
+        (_coro("test/0"), None, False),
+        (_coro("ceph-mon/1"), _coro("cpeh-mon/1"), False),
+        (_coro("test/0"), _coro(name="test/0"), True),
```

Comment:
This function passing got me surprised because this was the problem that I found to introduce the changes on #231

Now I discovered that the issue is comparing class methods that have async. E.g:

```python
class A:
    async def foo(self, time: int, msg:str):
        await asyncio.sleep(time)
        return msg

@pytest.mark.parametrize(
    "coro1, coro2, exp_result",
    [
        (A().foo(10, msg="hello"), A().foo(10, "hello"), True), # this unit test will fail

``` 

Not sure if we should spend time trying to fix this. I guess a unit test showing that this is not expected and a message on __eq__ method saying that it's not able to compare this situation on classes is enough to me.

Reply:
Actually my example had a problem and this also works:

```
class A:
    async def foo(self, time: int, msg:str):
        await asyncio.sleep(time)
        return msg

a = A()

@pytest.mark.parametrize(
    "coro1, coro2, exp_result",
    [
        (a.foo(10, "hello"), a.foo(10, msg="hello"), True),
```

So I don't know how to reproduce the error that I was finding....
In this case I'll approve it

Reply:
I was able to reproduce it....
If you go to `test_get_pause_unit_step` https://github.com/canonical/charmed-openstack-upgrader/blob/9e3a89a70c4f11518da459191a58488be42ebbfb/tests/unit/apps/test_base.py#L142-L157

and remove the kwargs the test will fail. Probably this is because of the model fixture that creates a `AsyncMock()` to `run_action` that might behave differently and it's not related to `compare_step_coroutines` itself.